test/unittest/utils/LocatorTests.cpp

```diff
@@ -1654,6 +1780,106 @@ TEST(LocatorListComparisonTests, locatorList_comparison)
+            {"www.eprosima.com", {{"154.56.134.194"}, {}}}, // Only IPv4
+            {"www.google.com", {{"216.58.215.164"}, {"2a00:1450:400e:803::2004"}}},
+            {"www.google.es", {{"142.250.184.3"}, {"2a00:1450:4003:808::2003"}}},
+            {"www.github.com", {{"140.82.121.4", "140.82.121.3"}, {}}},
+            {"docs.ros.org", {{}, {"2605:bc80:3010:104::8cd3:962"}}} // Only IPv6
+        };
+
+    for (auto const& address : addresses)
+    {
+        bool found_at_least_one = false;
+        std::pair<std::set<std::string>, std::set<std::string>> dns_result;
+
+        dns_result = IPLocator::resolveNameDNS(address.first);
+
+        for (auto ipv4 : address.second.first)
+        {
+            if (dns_result.first.find(ipv4) != dns_result.first.end())
+            {
+                found_at_least_one = true;
+                break;
+            }
+        }
+
+        if (!found_at_least_one)
+        {
+            for (auto ipv6 : address.second.second)
+            {
+                if (dns_result.second.find(ipv6) != dns_result.second.end())
+                {
+                    found_at_least_one = true;
+                    break;
+                }
+            }
+        }
+
+        // If it arrives here is that any correct ip has been found for one case
+        if (!found_at_least_one)
+        {
+            std::cout << "IP not found for domain: " << address.first << std::endl;
+        }
+        EXPECT_TRUE(found_at_least_one);
+    }
+}
+
+/*
+    This test uses the DNS and IPs onf the `resolve_name` test
+    to check that the locators are correctly formed
+*/
+TEST(LocatorDNSTests, dns_locator)
+{
+    std::map<std::string, std::string> dns_locator_to_address =
+        {
+            {"IPv4:[localhost]:1024", "127.0.0.1"},
+            {"IPv6:[localhost]:1024", "::1"},
+            {"IPv4:[www.eprosima.com]:1024", "154.56.134.194"},
+            {"IPv6:[www.eprosima.com]:1024", ""},   //invalid locator
+            {"IPv4:[www.eprosima.com]:1024", "154.56.134.194"},
+            {"IPv6:[www.eprosima.com]:1024", ""},   // invalid locator
+            {"IPv4:[www.google.com]:1024", "216.58.215.164"},
+            {"IPv6:[www.google.com]:1024", "2a00:1450:400e:803::2004"},
+            {"IPv4:[www.google.es]:1024", "142.250.184.3"},
+            {"IPv6:[www.google.es]:1024", "2a00:1450:4003:808::2003"},
+            {"IPv4:[docs.ros.org]:1024", ""},   // invalid locator
+            {"IPv6:[docs.ros.org]:1024", "2605:bc80:3010:104::8cd3:962"}
+        };
+
+    for (auto const& dns : dns_locator_to_address)
+    {
+        std::stringstream ss_dns(dns.first);
+        Locator_t locator;
+        ss_dns >> locator;
+        if (dns.second.empty())
+        {
+            EXPECT_EQ(LOCATOR_KIND_INVALID, locator.kind);
+        }
+        else
+        {
+            std::stringstream ss_locator;
+            ss_locator << locator;
+            EXPECT_EQ(dns.second, ss_locator.str());
+        }
+    }
+}
```

Comment:
I expect both DNS tests to be flakey, since the IPs returned by the DNS could change. How could we solve this? Are there any standard specified names and IPs that could be used?

Reply:
The only solution I can think of is to add these entries to the `/etc/hosts` (on POSIX) or `c:\windows\system32\drivers\etc\hosts` (on Windows).

However, this must be done in any machine running the tests. We can filter out the test during compilation unless `EPROSIMA_BUILD` is active. This reduces the chances of this test being run outside our CI (or our dev machines)